### PR TITLE
Recognize ternaryOp VectorAPI intrinsic and simplify using vcall

### PR DESCRIPTION
**Pull request by gita-omr**
- recognize new intrinsic
- delete using TR_UseVCall environment variable and specify opcodes
  using useVcallForVectorAPIOpcode method in VectorAPIExpansion.cpp

### CODE REVIEW

**Comment by gita-omr** [comment]
Fixed copyright and white spaces.

**Review by hzongaro — APPROVED** [review]
I think the changes look good.  Just a really minor question/comment.

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 1]
```diff
@@ -1295,30 +1295,56 @@ TR::Node *TR_VectorAPIExpansion::binaryIntrinsicHandler(TR_VectorAPIExpansion *o
...
+   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, mode, 3);
+   }
+
+bool TR_VectorAPIExpansion::useVcallForVectorAPIOpcode(TR::Compilation *comp, int32_t vectorAPIOpcode, vec_sz_t vectorLength)
+   {
+   // Check for opcodes that will use vcall temporarily.
+   // Add other specific opcodes here that you would like to prototype using vcall
+   // This is meant for prototyping only. Please do not enable when merging.
+#if 0
+   if ((vectorAPIOpcode == VECTOR_OP_FMA
```
[the remark this answers is not in the document]
Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1310]: Yeah, I know it's a little bit confusing and it's a good question. It's just gives an opportunity for somebody to implement FMA as a vcall in the codegen.  We already have some examples in the codegen on how to do that. Basically, it's meant to help common code and codegen work to proceed in parallel even if there is no vector IL opcode yet (eg. we don't have vfma yet).

**Comment by gita-omr** [comment]
Sorry, pushed wrong commit. Reverting to the original...

**Comment by gita-omr** [comment]
Pushed the original commit.

**Comment by vijaysun-omr** [comment]
Just so I have more context (and for future reference for looking at such PRs where we may want to use vcalls) what is the motivation behind using vcall vs not using it ? i.e. what is the situation in which it is warranted ?

**Comment by vijaysun-omr** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by gita-omr** [comment]
> Just so I have more context (and for future reference for looking at such PRs where we may want to use vcalls) what is the motivation behind using vcall vs not using it ? i.e. what is the situation in which it is warranted ?

I think I described it in some other PR but not in this one. vcall is a temporary solution until we have opcodes for all vector operations we need. For example, we have vadd opcode but not vfma. In order to proceed with the codegen work we can generate vcall to ternaryOp() intrinsic that will have FMA opcode as a parameter as well as other operands as arguments. This code should not be enabled in any release since it's not completely functionally correct. But it allows codegen and common work proceed in parallel. Said that, I hope we will have new vector IL opcodes implemented soon.

@midronij fyi

**Pull request merged**